### PR TITLE
Switch sale_filter_logic

### PR DESCRIPTION
**Pull request by wagnerlmichael**
draft test - to be deleted eventually

Had to switch order of my ias and mydec CTEs for this to work.

### CODE REVIEW

File: dbt/models/default/default.vw_pin_sale.sql [hunk 1]
```diff
@@ -106,17 +193,29 @@ unique_sales AS (
...
-            LAG(DATE_PARSE(SUBSTR(sales.saledt, 1, 10), '%Y-%m-%d')) OVER (
+            -- Recompute same_price_earlier_date using adjusted_sale_date
+            LAG(
+                CASE
+                    WHEN
+                        mydec_sales.mydec_date IS NOT NULL
+                        AND mydec_sales.mydec_date != DATE_PARSE(SUBSTR(sales.saledt, 1, 10), '%Y-%m-%d')
+                        THEN mydec_sales.mydec_date
+                    ELSE DATE_PARSE(SUBSTR(sales.saledt, 1, 10), '%Y-%m-%d')
+                END
```
Comment (wagnerlmichael) [inline dbt/models/default/default.vw_pin_sale.sql:204]: Grab sale in question.

File: dbt/models/default/default.vw_pin_sale.sql [hunk 2]
```diff
@@ -9,8 +9,8 @@ WITH town_class AS (
         CONCAT(
             leg.user1, SUBSTR(REGEXP_REPLACE(par.nbhd, '([^0-9])', ''), 3, 3)
         ) AS nbhd
-    FROM {{ source('iasworld', 'pardat') }} AS par
-    LEFT JOIN {{ source('iasworld', 'legdat') }} AS leg
+    FROM iasworld.pardat AS par
```
Comment (wrridgeway) [inline dbt/models/default/default.vw_pin_sale.sql:12]: Any particular reason to hard-code this?

**Comment by wrridgeway** [comment]
Alright can we just do some tests in here to make sure this is working as expected? Basically, everything should look the same as before except the re-defined filter, yeah?

**Comment by wagnerlmichael** [comment]
> @wagnerlmichael Alright, can we just do some tests in here to make sure this is working as expected? Basically, everything should look the same as before except the re-defined filter, yeah?

Yeah, and then once we check that is right I was going to implement the same logic in the PR where we incorporate this change.

**Comment by wagnerlmichael** [comment]
## sale_filter_same_sale_within_365
```sql 
SELECT count(*), sale_filter_same_sale_within_365
FROM "default"."vw_pin_sale"
group by sale_filter_same_sale_within_365
```
12668	true
2483405	false

```sql
SELECT count(*), sale_filter_same_sale_within_365 FROM
"z_ci_sale_filter_fix_default"."vw_pin_sale"
group by sale_filter_same_sale_within_365
```
13188	true
2482885	false

## Total counts

```sql
SELECT count(*) FROM
"z_ci_sale_filter_fix_default"."vw_pin_sale"
```
2496073

```sql
SELECT count(*)
FROM "default"."vw_pin_sale"
```

## Check observations per year
```sql
SELECT count(*), year
FROM "default"."vw_pin_sale"
group by year
```
2496073

```sql
SELECT count(*), year FROM
"z_ci_sale_filter_fix_default"."vw_pin_sale"
group by year
```

**Comment by wrridgeway** [comment]
@wagnerlmichael do we still need this PR?

**Comment by wagnerlmichael** [comment]
> @wagnerlmichael do we still need this PR?

Don't think so, I'll close it.

**Pull request closed without merging**